pennylane/spin/lattice.py

```diff
@@ -134,6 +142,17 @@ def __init__(
 
         self.edges_indices = [(v1, v2) for (v1, v2, color) in self.edges]
 
+        if custom_nodes is not None:
+            for node in custom_nodes:
+                if node[0] > self.n_sites:
+                    raise ValueError(
+                        "The custom node has an index larger than the number of sites."
+                    )
+                if node[0] < 0:
+                    raise ValueError("The custom node has an index smaller than 0.")
```

Comment:
Ideally we can combine these into one.

Reply:
Hmm, I mean we could, but it looks fine to me this way as well.

Reply:
Just a suggestion, not a blocker. 👍